server/webhook.go

```diff
@@ -178,6 +177,19 @@ func (s *Server) filterRepository(ctx context.Context, webhookEndpointID string,
 	return filteredRepos, nil
 }
 
+func validateWebhookEventBranch(pushEventRef string, repo *api.Repository) (bool, error) {
```

Comment:
s/validateWebhookEventBranch/isWebhookEventBranch

And pass in repo.BranchFilter and repo.BranchID instead of repo.